assets/src/data/persistence/extrinsic.ts

```diff
@@ -100,7 +125,19 @@ export const updateGlobalUserState = async (
   updates: { [topKey: string]: { [key: string]: any } },
   useLocalStorage = false,
 ) => {
-  /* console.log('updateGlobalUserState called', { updates, useLocalStorage }); */
+  //Lets update the cache with latest changes.
+  const topLevelKeys = Object.keys(updates);
+  userStateCache['timestamp'] = Date.now();
+  topLevelKeys.forEach((topKey: any) => {
```

Comment:
this logic is basically repeated from the internal one, so probably should factor it out into a shared method